Java/core/src/main/java/com/amazon/randomcutforest/store/LeafStore.java

```diff
@@ -67,51 +67,40 @@ public int addLeaf(int parentIndex, int pointIndex, int mass) {
         this.parentIndex[index] = (short) parentIndex;
         this.mass[index] = (short) mass;
         this.pointIndex[index] = pointIndex;
-        return index + super.capacity;
-    }
-
-    @Override
-    public boolean isLeaf(int index) {
-        return index >= super.capacity;
+        return index;
     }
 
     @Override
     public void setParent(int index, int parent) {
-        parentIndex[index - super.capacity] = (short) parent;
+        parentIndex[index] = (short) parent;
```

Comment:
I think we will have a "large" node store/leafstore if the sampleSize is large (for large trees). Otherwise we have these small trees (but a large number of them). The trees always see int (irrespective of large/small size). The ILeafStore/INodeStore interfaces are int.